README.md

```diff
@@ -11,19 +11,81 @@ $ go get github.com/daixiang0/gci
+                                   NL | NewLine - Prints an empty line
+                                   Prefix(gitlab.com/myorg) | pkgPrefix(gitlab.com/myorg) - Groups all imports with the specified Prefix. Imports will be matched to the longest Prefix.
+                                   Std | Standard - Captures all standard packages if they do not match another section
+                                    (default [Standard,Default])
+  -x, --SectionSeparator strings   SectionSeparators are inserted between Sections (default [NewLine])
+  -h, --help                       help for write
+```
 
 ```shell
-$ gci -h
-usage: gci [flags] [path ...]
-  -d	display diffs instead of rewriting files
-  -local string
-    	put imports beginning with this string after 3rd-party packages, only support one string
-  -w	write result to (source) file instead of stdout
+$ gci diff -h
+Diff prints a patch in the style of the diff tool that contains the required changes to the file to make it adhere to the specified formatting.
+
+Usage:
+  gci diff path... [flags]
+
+Flags:
+      --NoInlineComments           Drops inline comments while formatting
+      --NoPrefixComments           Drops comment lines above an import statement while formatting
+  -s, --Section strings            Sections define how inputs will be processed. Section names are case-insensitive and may contain parameters in (). A section can contain a Prefix and a Suffix section which is delimited by ":". These sections can be used for formatting and will only be rendered if the main section contains an entry.
+                                   Comment(your text here) | CommentLine(your text here) - Prints the specified indented comment
+                                   Def | Default - Contains all imports that could not be matched to another section type
+                                   NL | NewLine - Prints an empty line
+                                   Prefix(gitlab.com/myorg) | pkgPrefix(gitlab.com/myorg) - Groups all imports with the specified Prefix. Imports will be matched to the longest Prefix.
+                                   Std | Standard - Captures all standard packages if they do not match another section
+                                    (default [Standard,Default])
+  -x, --SectionSeparator strings   SectionSeparators are inserted between Sections (default [NewLine])
+  -d, --debug                      Enables debug output from the formatter
+  -h, --help                       help for diff
 ```
+Support for the old CLI style is still present if you do not specify the subcommands. The only difference is that `--local` requires two dashes now.
 
 ## Examples
 
-Run `gci -w -local github.com/daixiang0/gci main.go` and you will handle following cases.
+Run `gci  write --Section std --Section nl:default --Section "NewLine:Prefix(github.com/daixiang0/gci)" main.go` and you will handle following cases.
```

Comment:
The values of those options are defaulted to ensure the default behavior doesn't change?

Reply:
Thanks for bringing this up, I have noticed that the flag format noted in the Readme was from an older version and have fixed that.
Both of the examples have corresponding unit tests:
https://github.com/ngehrsitz/gci/blob/section-support/pkg/gci/internal/testdata/simple-case.in.go
https://github.com/ngehrsitz/gci/blob/section-support/pkg/gci/internal/testdata/with-alias.in.go
Thus we can be sure that the cases listed there will continue to work!

Reply:
Pretty cool!